frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -674,18 +674,19 @@ function ReaderHighlight:updateHighlight(page, index, side, direction, move_by_c
         if updated_highlight_end then
             local order = self.ui.document:compareXPointers(highlight_beginning, updated_highlight_end)
             if order and order > 0 then -- only if end did not go back past beginning
-                self.view.highlight.saved[page][index].pos1 = updated_highlight_end
+                highlight.pos1 = updated_highlight_end
             end
         end
     end
 
-    local new_beginning = self.view.highlight.saved[page][index].pos0
-    local new_end = self.view.highlight.saved[page][index].pos1
+    local new_beginning = highlight.pos0
+    local new_end = highlight.pos1
     local new_text = self.ui.document:getTextFromXPointers(new_beginning, new_end)
     local new_chapter = self.ui.toc:getTocTitleByPage(new_beginning)
-    self.view.highlight.saved[page][index].text = cleanupSelectedText(new_text)
-    self.view.highlight.saved[page][index].chapter = new_chapter
-    local new_highlight = self.view.highlight.saved[page][index]
+    highlight.text = cleanupSelectedText(new_text)
+    highlight.chapter = new_chapter
+    local new_highlight = highlight
+    self:_insertHighlight(page, new_highlight)
     self.ui.bookmark:updateBookmark({
         page = highlight_beginning,
         datetime = highlight_time,
```

Comment:
I guess you don't need `local new_highlight` (which would be just an alias).
Also, `highlight` is the same table all around, removed from `self.view.highlight.saved[page]` and inserted back, and passed to `self.ui.bookmark:updateBookmark()`.
It was already the case before, so no real need it seems to "deepCopy" the table in a `new_highlight` different table (which could justify a new name, as it would be a different object).

Reply:
Done.